src/overlays/actors/ovl_Obj_Mure2/z_obj_mure2.c

```diff
@@ -30,43 +51,204 @@ ActorInit Obj_Mure2_InitVars = {
     /**/ NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_809615F4[] = {
-    ICHAIN_F32(uncullZoneForward, 100, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneScale, 2100, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_STOP),
+static f32 sActivationRangesSq[] = {
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_BUSH_RING
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_ROCK_RING
 };
 
-#endif
-
-extern InitChainEntry D_809615F4[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960CF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960E0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960F0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80961018.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_8096104C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809611BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809612BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/ObjMure2_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613B0.s")
+static f32 sDeactivationRangesSq[] = {
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_BUSH_RING
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613C4.s")
+static s16 sChildCounts[] = {
+    OBJ_MURE2_CHILD_COUNT_BUSH_RING,      // OBJMURE2_CHILDTYPE_BUSH_RING
+    OBJ_MURE2_CHILD_COUNT_BUSH_SCATTERED, // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    OBJ_MURE2_CHILD_COUNT_ROCK_RING,      // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613E8.s")
+static s16 sActorIds[] = {
+    ACTOR_EN_KUSA,
+    ACTOR_EN_KUSA,
+    ACTOR_EN_ISHI,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613FC.s")
+static VecPolarS sScatteredBushSpawnInfo[] = {
+    { 40, 0x0666 }, { 40, 0x2CCC }, { 40, 0x5999 }, { 40, 0x8666 }, { 20, 0xC000 }, { 80, 0x1333 },
+    { 80, 0x4000 }, { 80, 0x6CCC }, { 80, 0x9333 }, { 80, 0xACCC }, { 80, 0xC666 }, { 60, 0xE000 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_8096147C.s")
+typedef void (*ObjMure2SpawnPosFunc)(Vec3f[12], ObjMure2*);
```

Comment:
Correct me if I'm wrong, but this `12` seems to be the `OBJ_MURE2_CHILD_MAX` constant you made?
```suggestion
typedef void (*ObjMure2SpawnPosFunc)(Vec3f[OBJ_MURE2_CHILD_MAX], ObjMure2*);
```